assets/js/search-page.js

```diff
@@ -27,30 +28,85 @@ function initialize () {
   const pathname = window.location.pathname
   if (pathname.endsWith('/search.html') || pathname.endsWith('/search')) {
     const query = getQueryParamByName('q')
-    search(query)
+    const queryType = getQueryParamByName('type')
+    search(query, queryType)
   }
 }
 
-async function search (value) {
+async function search (value, queryType) {
   if (isBlank(value)) {
     renderResults({ value })
   } else {
     setSearchInputValue(value)
 
-    const index = await getIndex()
-
     try {
-      // We cannot match on atoms :foo because that would be considered
-      // a filter. So we escape all colons not preceded by a word.
-      const fixedValue = value.replaceAll(/(\B|\\):/g, '\\:')
-      const results = searchResultsToDecoratedSearchItems(index.search(fixedValue))
+      let results = []
+      const searchNodes = getSearchNodes()
+
+      if (['related', 'latest'].includes(queryType) && searchNodes.length > 0) {
+        results = await remoteSearch(value, queryType, searchNodes)
```

Comment:
Can we can a race condition here? When the previous request returns after the current request and updates the items to stale results?

Might be helpful: https://github.com/ruslandoga/ex_doc/pull/1

Reply:
Thanks. I'm sure you're right. As you can probably tell it's been almost a decade since I wrote any JavaScript so I'm still getting the hang of the new idioms.

Reply:
So looking at the code more carefully, it appears that the `search` function is only called on page load, so should only be run once in the page's lifecycle.

Reply:
Additionally, the search result handlebars template takes care of whether any results were actually returned.